### PR TITLE
adds lists for r lang

### LINKED ISSUE #577: Add R language
# Language you would like added
R  
[R project](https://www.r-project.org/about.html)

# Information on the language
This is a language primarily for stats that has grown into a popular tool for data science that is being used to create webpages and books, so it is more flexible than originally imagined.

# Disclaimer
If you would like to work on this issue, please comment here so that you are assigned.
I'd like to work on this issue

### PR DESCRIPTION
**Pull request by monkeywithacupcake**
## What GitHub issue does this PR apply to?

Resolves #577

## What changed and why?

 added lists.json for R because it was not there

## Checklist

   

- [X] I claimed any associated issue(s) and they are not someone else's
- [X] I have looked at documentation to ensure I made any revisions correctly
- [X] I tested my changes locally to ensure they work
- [ ] (If editing Django) I have added or edited any appropriate unit tests for my changes

## Any additional comments or things to be aware of while reviewing?

In some places, I really wanted to add caveats that you probably would not use a "list" in r for this. So, I expect we will have future edits.

### CODE REVIEW

File: web/thesauruses/r/4/lists.json [hunk 1]
```diff
@@ -0,0 +1,511 @@
...
+          "1"
+        ]
+      },
+      "unordered_mutable_list_can_be_appended": {
+        "name": "Can you append to it?",
+        "code": [
+          "mylist <- list(a = 1, b = 2, c = 3)",
+          "mylist <- append(mylist, list(d = 4))"
+        ]
+      },
```
Comment (geekygirlsarah) [inline web/thesauruses/r/4/lists.json]: I think this is intended to be a yes/no kind of question (as how you append is below)

File: web/thesauruses/r/4/lists.json [hunk 2]
```diff
@@ -0,0 +1,511 @@
...
+        "code": [
+            "mylist <- list(a = 1, b = 2, c = 3)",
+            "mylist <- append(mylist, list(x = 42), length(mylist))"
+        ]
+      },
+      "insert_into_middle_of_unordered_mutable_list": {
+        "name": "Insert element in middle",
+        "code": [
+            "mylist <- list(a = 1, b = 2, c = 3)",
+            "mylist <- append(mylist, list(x = 42), 2)"
```
Comment (geekygirlsarah) [inline web/thesauruses/r/4/lists.json:179]: Is it safe to assume the `,2` is to insert in position 2? But also does this start at 0 or at 1, or is it based on the numbering from the list on the line above? Maybe add a `comment:` to clarify?

**Review by geekygirlsarah — CHANGES_REQUESTED** [review]
Two tiny clarifications and I think this is good to merge!

**Comment by monkeywithacupcake** [comment]
thanks for that @geekygirlsarah  - the yes/no questions kept tripping me up!

**Review by geekygirlsarah — APPROVED** [review]
Sweet, I'll merge this in. Thank you for adding more R to the site!

**Pull request merged**